src/main/java/stream/flarebot/flarebot/util/general/GeneralUtils.java

```diff
@@ -191,194 +401,38 @@ public static AudioItem resolveItem(Player player, String input) throws IllegalA
-                List<TextChannel> tcs = wrapper.getGuild().getTextChannelsByName(channelArg, true);
-                if (!tcs.isEmpty()) {
-                    return tcs.get(0);
+    /**
+     * Checks if a perm is valid.
+     *
+     * @param perm The perm to check.
+     * @return if it's valid.
+     */
+    public static boolean validPerm(String perm) {
+        if (perm.equals("*") || perm.equals("flarebot.*")) return true;
+        if (perm.startsWith("flarebot.") && perm.split("\\.").length >= 2) {
+            perm = perm.substring(perm.indexOf(".") + 1);
+            String command = perm.split("\\.")[0];
+            for (Command c : FlareBot.instance().getCommandManager().getCommands()) {
+                if (c.getCommand().equalsIgnoreCase(command) && c.getType() != CommandType.SECRET) {
+                    return true;
                 }
             }
-            return null;
-        }
-    }
-
-    public static void joinChannel(TextChannel channel, Member member) {
-        GuildVoiceState vs = channel.getGuild().getSelfMember().getVoiceState();
-        GuildVoiceState memberVS = member.getVoiceState();
-        if (memberVS.getChannel() == null)
-            return; // They aren't in a VC so we can't join that.
-        if (vs.getChannel() != null && vs.getChannel().getIdLong() == member.getVoiceState().getChannel().getIdLong())
-            return; // Already in the same VC as the user.
-
-        if (channel.getGuild().getSelfMember()
-                .hasPermission(memberVS.getChannel(), Permission.VOICE_CONNECT) &&
-                channel.getGuild().getSelfMember()
-                        .hasPermission(memberVS.getChannel(), Permission.VOICE_SPEAK)) {
-            if (memberVS.getChannel().getUserLimit() > 0 && member.getVoiceState().getChannel()
-                    .getMembers().size()
-                    >= member.getVoiceState().getChannel().getUserLimit() && !member.getGuild().getSelfMember()
-                    .hasPermission(member
-                            .getVoiceState()
-                            .getChannel(), Permission.MANAGE_CHANNEL)) {
-                MessageUtils.sendErrorMessage("We can't join :(\n\nThe channel user limit has been reached and we don't have the 'Manage Channel' permission to " +
-                        "bypass it!", channel);
-                return;
-            }
-            channel.getGuild().getAudioManager().openAudioConnection(member.getVoiceState().getChannel());
-        } else {
-            MessageUtils.sendErrorMessage("I do not have permission to " + (!channel.getGuild().getSelfMember()
-                    .hasPermission(member.getVoiceState()
-                            .getChannel(), Permission.VOICE_CONNECT) ?
-                    "connect" : "speak") + " in your voice channel!", channel);
         }
+        return false;
     }
 
+    /**
+     * Orders a Collection
+     *
+     * @param strings The Collection to order
```

Comment:
How does it order it? Does it only take strings? Surely that variable should be named better